### PR TITLE
new package to be added: NuT

### PR DESCRIPTION
**Pull request by jiseung**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nut/package.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    # must be built with clang
+    conflicts('%gcc')
+    conflicts('%intel')
+    conflicts('%pgi')
+    conflicts('%xl')
+    conflicts('%nag')
+
+    def cmake_args(self):
+        env['RANDOM123_DIR'] = '{0}'.format(self.spec['random123'].prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: There's no need to convert this to a string, it is already a string.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        env['CC'] = 'cc'
+        env['CXX'] = 'c++'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: These are already set by Spack.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        self.build_targets.extend(
+            ['VERBOSE=on -j 4 2>&1 | tee -a make.out'])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: You should set this directly outside of `cmake_args`.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        args = ['..']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: This is already included by default in `cmake_args`.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        return args
+
+    def install(self, spec, prefix):
+        install('README.md', prefix)
+        install_tree(
+            join_path(self.build_directory, 'test'),
+            join_path(prefix, 'test'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: You can replace this with:
```python
install_tree('test', prefix.test)
```

File: var/spack/repos/builtin/packages/nut/package.py (hunk 4 above)
Comment (jiseung) [inline var/spack/repos/builtin/packages/nut/package.py]: Should I define the environment variable 'RANDOM123_DIR' outside of `cmake_args` on line 55?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: You should define it in `setup_environment`.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    conflicts('%gcc')
+    conflicts('%intel')
+    conflicts('%pgi')
+    conflicts('%xl')
+    conflicts('%nag')
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('RANDOM123_DIR', self.spec['random123'].prefix)
+    
+    build_targets = ['VERBOSE=on -j 4 2>&1 | tee -a make.out']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: This probably needs to be a list of strings, not one big string. Otherwise it is equivalent to putting quotes around all of these arguments on the command line, they will be grouped in a single string.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: Also, you don't want to specify the parallelism. It's already set by Spack and you want to be able to control it.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: And there's no need to redirect the output of the build. Spack already redirects it to `spack-build.out`.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    conflicts('%gcc')
+    conflicts('%intel')
+    conflicts('%pgi')
+    conflicts('%xl')
+    conflicts('%nag')
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('RANDOM123_DIR', self.spec['random123'].prefix)
+ 
+    build_targets = ['VERBOSE=on -j 4 2>&1 | tee -a make.out']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: Adding these comments again because they seem to be hidden by another commit:

This probably needs to be a list of strings, not one big string. Otherwise it is equivalent to putting quotes around all of these arguments on the command line, they will be grouped in a single string.

Also, you don't want to specify the parallelism. It's already set by Spack and you want to be able to control it.

And there's no need to redirect the output of the build. Spack already redirects it to `spack-build.out`.

Basically, `VERBOSE=on` might be useful but the rest aren't.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+    homepage = "https://github.com/lanl/NuT"
+    url      = ""
+    tags     = ['proxy-app']
+
+    version(
+        'serial', git='https://github.com/lanl/NuT.git',
+        branch='master')
+    version(
+        'omp', git='https://github.com/lanl/NuT.git',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: I would call this version `openmp`.

File: var/spack/repos/builtin/packages/nut/package.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    conflicts('%nag')
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('RANDOM123_DIR', self.spec['random123'].prefix)
+ 
+    build_targets = ['VERBOSE=on -j 4 2>&1 | tee -a make.out']
+
+    def install(self, spec, prefix):
+        install('README.md', prefix)
+        install_tree('test', prefix.test)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nut/package.py]: The `test` directory is the only thing you want to install? Surely there's more to the package than that.
